ecosystem/op-deployer.md

```diff
@@ -0,0 +1,163 @@
+# Purpose
+
+Introduce `op-deployer`, an end-to-end chain deployment tool, and get buy-in on its architecture and implementation.
+
+## Goals
+
+- Provide a canonical way to deploy chains.
+- Enable deploying multiple L2s to a single L1 as part of a single logical Superchain.
```

Comment:
What do you think about "Deploying multiple superchains to a single L1" as another goal? For example the devnet and testnet both have superchains on L1 Sepolia, so this should help with spinning up devnets. It also helps validate the architecture is sufficiently modular

Reply:
The tooling should support that use case. I imagine you'd do that by defining another deployment intent, and running that through the tooling. Is there a use case for defining multiple superchains _and_ multiple L2s in the same deployment intent?

Reply:
I'm not clear on what exactly a deployment intent is and what execution of it does. Can it be scoped to a single step of the pipeline? What part of the intent file maps to a pipeline step, or is it one intents file per step?

Either way, yes a separate invocation with a different deployment intent to support this seems ok to me

Reply:
A deployment intent is a declarative way of describing how your deployment will look. The tooling then looks at the intent and decides what changes it needs to make on L1/l2 to make the real deployment match the intent. It's like Terraform, but for blockchains. I can clarify this more in the doc.

To your questions specifically:

> Can it be scoped to a single step of the pipeline?

No, the deployer would decide which steps it needs to run based on the changes it sees to the deployment intent.

> What part of the intent file maps to a pipeline step, or is it one intents file per step?

There is one intent per file, defined in the `[intent]` stanza.

Reply:
Will op-deployer support incremental deployments (i.e., only deploy the minimal set of components that were modified as a result of an intent mutation)?